### DISCUSSION TITLE
Empty --config-setting causing problems with meson-python build

### DISCUSSION THREAD

**Discussion by ianthomas23** [pypa/cibuildwheel#1471] [General]
I am in the process of porting an existing Python/C++ project that uses `cibuildwheel` to the `meson-python` build backend (https://github.com/contourpy/contourpy/pull/183). This is progressing well but there is one problem when using `cibuildwheel`.

The build verbosity flag passed to `meson-python` has a specific form of a `config-setting` that is `compile-args=-v`. This is easily achieved in `cibuildwheel` using `CIBW_CONFIG_SETTINGS`. Hence I do not need to use `CIBW_BUILD_VERBOSITY` and I leave it unspecified. But when I do this I end up with a `build` command of this form:

```python
python -m build /project --wheel --config-setting=compile-args=-v --config-setting=
```
The problem is the empty `--config-setting=` at the end, which `meson-python` errors out on. The GHA run producing this is https://github.com/ianthomas23/contourpy/actions/runs/4715025410/jobs/8361738715 using `cibuildwheel` 2.12.1.

I figure that ideally `cibuildwheel` wouldn't emit the empty `--config-setting`. On naively looking at the `cibuildwheel` source code for the first time I think a possible fix is here

https://github.com/pypa/cibuildwheel/blob/7c9b837780c0a415a540abff18c986ca8aa79b7f/cibuildwheel/linux.py#L263-L265

not appending to the `extra_flags` if the `verbosity_setting` is empty. But I assume there must be other places for a similar change for non-linux platforms.

Of course it could be argued that `meson-python` should tolerate the empty setting that is currently produced.

**Comment by henryiii** [comment]
That line (edit: two lines) looks suspect as is and I would not argue to make meson-python ignore empty settings. :)

**Reply by henryiii** [reply]
Also, is this really correct? This is not generic to all back ends, and is not the same thing as enabling verbose mode on pip. Build defaults to verbose mode, basically. I think it should probably be removed.

**Reply by joerick** [reply]
Agreed, this looks like a bug
